Samples/OLED1Xplained/OLED1.lpr

```diff
@@ -0,0 +1,152 @@
+  Library General Public License for more
+
+  This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the FPC modified GNU Library General Public
+  License for more details.
+}
+
+{< MBED application shield demo program. }
+
+{$INCLUDE MBF.Config.inc}
+
+{.$define DEBUG}
+
+uses
+  heapmgr, // this is needed for SAMD10, but keep it in place for the others also !!!!
+  MBF.BitHelpers,
+  MBF.__CONTROLLERTYPE__.SystemCore,
+  MBF.__CONTROLLERTYPE__.GPIO,
+  MBF.__CONTROLLERTYPE__.SPI,
+  MBF.Displays.OLED1Xplained;
+
+const
+  {$if defined(has_samd20_xplained_pro) or defined(has_samd21_xplained_pro) or defined(has_samc21_xplained_pro)}
+    PinLED     = LED0;
+    PinButton  = SW0;
+
+    PortSPI    = 5;
+    PortSPIMux = Ord(TPinMux.MuxD);
+    PinMOSIMux = PortSPIMux;
+    PinMISOMux = Ord(TPinMux.None);
+    PinSCKMux  = PortSPIMux;
+
+    PinReset   = EXT3_PIN_10;
+    PinMISO    = TNativePin.None;
+    PinSS      = EXT3_PIN_15;
+    PinMOSI    = EXT3_PIN_16;//PAD[2]
+    PinSCK     = EXT3_PIN_18;//PAD[3]
+    PinA0      = EXT3_PIN_5;
+  {$endif}
+
+var
+  Display:TDisplay;
+  SPI1:TSPI_Registers;
+  TurnedOn:boolean;
+  i:byte;
+
+procedure ToggleLed;
+begin
+  TurnedOn:=NOT TurnedOn;
+  if TurnedOn then
+    GPIO.PinValue[PinLED]   := 1
+  else
+    GPIO.PinValue[PinLED]   := 0;
+end;
+
+begin
+  SystemCore.Initialize;
+
+  GPIO.Initialize;
+
+  // Switch on LED pin for output.
+  GPIO.PinMode[PinLED]  := TPinMode.Output;
+  GPIO.PinValue[PinLED] := 0;
+
+  SPI1.Initialize(PortSPI,100000,0,1);
+
+  GPIO.PinMode[PinSS]     := TPinMode.Output;
+  GPIO.PinValue[PinSS]    := 1;
+  GPIO.PinMode[PinReset]  := TPinMode.Output;
+  GPIO.PinValue[PinReset] := 1;
+  GPIO.PinMode[PinA0]     := TPinMode.Output;
+  GPIO.PinValue[PinA0]    := 1;
+
+  GPIO.PinMux[PinMOSI]    := TPinMux(PinMOSIMux);
+  GPIO.PinMux[PinSCK]     := TPinMux(PinSCKMux);
+  if (PinMISO<>TNativePin.None) then {%H-}GPIO.PinMux[PinMISO] := TPinMux(PinMISOMux);
+
+  Display.Initialize(SPI1,PinA0,PinReset,ScreenSize128x32x1);
+  Display.PinCS:=PinSS;
+
+  Display.InitSequence;
+
+  // not enough flash ... :-(
+  {$ifndef samd10}
+  Display.GetDemo1;
+  ToggleLed;
+  SystemCore.Delay(2500);
+
+  Display.InvertScr(true);
+  Display.GetDemo2;
+  ToggleLed;
+  SystemCore.Delay(2500);
+  {$endif}
+
+  Display.ClrScr;
+  Display.DrawString(5,0,'MBED 0.16 demo.');
+  Display.DrawString(5,1,'Powered by FPC.');
+  Display.DrawString(5,2,'Powered by Lazarus.');
+  Display.DrawString(5,3,'By Michael / Alfred.');
+  Display.Present;
+  SystemCore.Delay(3000);
+
+  TurnedOn:=false;
+
+  GPIO.PinValue[PinLED] := 1;
+
+  Display.InvertScr(false);
+  Display.ClrScr;
+  Display.Present;
+  for i:=0 to (ScreenSize128x32x1.Width-1) do
+  begin
+    Display.Pixel[i,0]:=1;
+    Display.Present;
+  end;
+  for i:=0 to (ScreenSize128x32x1.Height-1) do
```

Comment:
Why make this so complicated? ScreenInfo is a public property of (my) TCustomDisplay, better use this instead...

Reply:
For sure !!! :-)
My mistake.